pkg/util/resourcequota/calculator_test.go

```diff
@@ -464,3 +466,203 @@ func Test_vmValidator_containsEnoughResourceQuotaToStartVM(t *testing.T) {
 		})
 	}
 }
+
+// when setting `additional-guest-memory-overhead-ratio` changes, the VM might reach the limit and can't start
+// tune those two parameters accordingly
+func Test_vmValidator_containsEnoughResourceQuotaToStartVM_WithChangedOverheadRatio(t *testing.T) {
+	generateRQ := func(cpu, mem int64) kubevirtv1.ResourceRequirements {
+		return kubevirtv1.ResourceRequirements{
+			Limits: map[corev1.ResourceName]resource.Quantity{
+				corev1.ResourceMemory: *resource.NewQuantity(mem*memory1Gi, resource.DecimalSI),
+				corev1.ResourceCPU:    *resource.NewQuantity(cpu, resource.DecimalSI),
+			},
+		}
+	}
+
+	nsrq := v3.NamespaceResourceQuota{
+		Limit: v3.ResourceQuotaLimit{
+			LimitsCPU:    "5000m",
+			LimitsMemory: "5120Mi", // 5Gi
+		},
+	}
+
+	tests := []struct {
+		name          string
+		vm            *kubevirtv1.VirtualMachine
+		nsrq          *v3.NamespaceResourceQuota
+		rq            *corev1.ResourceQuota
+		overheadRatio string
+		wantErr       bool
+	}{
+		{
+			name: "VM can start as quota is enough with default overhead-ratio 1.5",
+			vm: &kubevirtv1.VirtualMachine{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "vm1",
+					Namespace: testNS,
+				},
+				Spec: kubevirtv1.VirtualMachineSpec{
+					Template: &kubevirtv1.VirtualMachineInstanceTemplateSpec{
+						Spec: kubevirtv1.VirtualMachineInstanceSpec{
+							Domain: kubevirtv1.DomainSpec{
+								Resources: generateRQ(4, 4), // 4core, 4Gi
+							},
+						},
+					},
+				},
+			},
+			nsrq: &nsrq,
+			rq: &corev1.ResourceQuota{
```

Comment:
Similar to the VM suggestion we can extract and put as parameters to function the things that change in the spec, reducing the strain on understanding the test, reading function header is easier than going through object body IMHO